match_minima.py

```diff
@@ -0,0 +1,787 @@
+
+    # add legend, set plot limits, add grid
+    plt.legend(bbox_to_anchor=(0.96, 1), loc=2, prop={'size': 18})
+    plt.xlim(-1, ref_nconfs + 1)
+    ax.set_yticks(
+        np.arange(int(round(floor)) - 2,
+                  int(round(ceiling)) + 2))
+    plt.grid()
+
+    # save and close figure
+    plt.savefig(figname, bbox_inches='tight')
+    #plt.show()
+    fig.clear()
+    plt.close(fig)
+
+
+def match_minima(in_dict, rmsd_cutoff):
+    """
+    For different methods, match the conformer minima to those of the reference
+        method. Ex. Conf G of reference method matches with conf R of method 2.
+
+    Parameters
+    ----------
+    in_dict : OrderedDict
+        dictionary from input file, where key is method and value is dictionary
+        first entry should be reference method
+        in sub-dictionary, keys are 'sdfile' and 'sdtag'
+    rmsd_cutoff : float
+        cutoff above which two structures are considered diff conformers
+
+    Returns
+    -------
+    mol_dict : dict of dicts
+        mol_dict['mol_name']['energies']
+        = [[file1: conf1 conf2] [file2: conf1 conf2]]
+
+    """
+
+    # nested dictionary: 1st layer of mol names, 2nd layer of method energies
+    mol_dict = {}
+
+    # get first filename representing the reference geometries
+    sdf_ref = list(in_dict.values())[0]['sdfile']
+
+    # assess each file against reference
+    for ff_label, ff_dict in in_dict.items():
+        sdf_query = ff_dict['sdfile']
+        sdf_tag = ff_dict['sdtag']
+
+        # load molecules from open reference and query files
+        print("\n\nOpening reference file %s" % sdf_ref)
+        mols_ref = read_mols(sdf_ref)
+
+        print("Opening query file %s for [ %s ] energies" % (sdf_query, ff_label))
+        mols_query = read_mols(sdf_query)
+
+        # loop over each molecule in reference and query files
+        for rmol in mols_ref:
+            mol_name = rmol.GetTitle()
+            ref_nconfs = rmol.NumConfs()
+            run_match = False
+
+            for qmol in mols_query:
+
+                # same mol titles should mean same molecular identity;
+                # when same molecular identity found, break out of loop to
+                # start matching conformers
+                if rmol.GetTitle() == qmol.GetTitle():
+                    run_match = True
+                    break
+
+            # create entry for this mol in mol_dict if not already present
+            # energies [i][j] will be 2d list of ith file and jth conformer
+            if mol_name not in mol_dict:
+                mol_dict[mol_name] = {}
+                mol_dict[mol_name] = {'energies': [], 'indices': [], 'ref_nconfs': []}
+
+            # no same molecules were found bt ref and query files
+            # for N reference minima of each mol, P matching indices for each ref minimia
+            if not run_match:
+                print(f"No {mol_name} molecule found in {sdf_query}")
```

Comment:
For these sorts of statements, I find it helpful to put `{mol_name}` in quotes, since molecule names can be all sorts of things, so debugging an error where there's an extra space in the name (eg. `No cyclohexane  molecule found in mols.sdf` --> `No 'cyclohexane ' molecule found in 'mols.sdf`') or no mol name at all (`No  molecule found in mols.sdf` --> `No '' molecule found in 'mols.sdf'`)

Reply:
I very much agree with Jeff here.